lib/straight/exchange_rate_adapters/average_rate_adapter.rb

```diff
@@ -0,0 +1,22 @@
+module Straight
+  module ExchangeRate
+
+    class AverageRateAdapter
+
+      # Takes exchange rate adapters instances as an arguments
+      def initialize(*adapter_instances)
+        @rates = []
+        adapter_instances.each do |adapter|
+          @rates << adapter.rate_for('USD')
+        end
+      end
+
+      def calculate_avg_rate
+        @rates.compact!
+        @rates.inject {|sum, rate| sum + rate} / @rates.size
```

Comment:
Now, you have to make sure here, that if one adapter raises `CurrencyNotSupported` exception, we rescue it and ignore this adapter, relying on data from others. If, however, all of the adapters raise this exception and no exchange rate is received from at least one adapter, then we'll have to raise it!